src/audio/copier/copier.c

```diff
@@ -886,17 +864,11 @@ static int copier_prepare(struct comp_dev *dev)
 		return PPL_STATUS_PATH_STOP;
 
 	for (i = 0; i < cd->endpoint_num; i++) {
-		if (dev->ipc_config.type == SOF_COMP_HOST && !cd->ipc_gtw) {
-			ret = comp_set_state(cd->endpoint[i], COMP_TRIGGER_PREPARE);
-			if (ret < 0)
-				return ret;
-			if (ret == COMP_STATUS_STATE_ALREADY_SET)
-				return PPL_STATUS_PATH_STOP;
-
+		if (dev->ipc_config.type == SOF_COMP_HOST && !cd->ipc_gtw)
 			ret = host_zephyr_prepare(cd->hd);
```

Comment:
again - no need to do this in the loop, even though it's only executed once (or add a `break` or a check for `endpoint_num == 1`)

Reply:
moved out.